hw/misc/hermes.c

```diff
@@ -399,18 +497,61 @@ static uint64_t hermes_bar0_read(void *opaque, hwaddr addr, unsigned size)
             val &= 0xFFFF;
             break;
         }
+    } else if (addr >= 0x1000 &&
+               addr + size <= 0x1000 +
+               sizeof(struct hermes_bar0_command) * UBPF_ENGINES) {
+        ptr = (uint32_t *) &((uint8_t *) &bar0_init.commands)[addr];
+        val = *ptr;
```

Comment:
If `size != 4`, then we would be reading more than requested. I think the easiest solution is just to move the `switch (size)` to right before the return.